### PR TITLE
internal recode cna alt func

### PR DESCRIPTION
**Pull request by hfuchs5**
**What changes are proposed in this pull request?**
Separate branch with .recode_cna_alterations() function.

**If there is an GitHub issue associated with this pull request, please provide link.**
#178

--------------------------------------------------------------------------------

  Reviewer Checklist (if item does not apply, mark is as complete)

- [ ] PR branch has pulled the most recent updates from main branch. Ensure the pull request branch and your local version match and both have the latest updates from the main branch.
- [ ] If a new function was added, function included in `_pkgdown.yml`
- [ ] If a bug was fixed, a unit test was added for the bug check
- [ ] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [ ] Code coverage is suitable for any new functions/features. Review coverage with `withr::with_envvar(new = c("NOT_CRAN" = "true"), covr::report())`. Begin in a fresh R session without any packages loaded.
- [ ] R CMD Check runs without errors, warnings, and notes
- [ ] `usethis::use_spell_check()` runs with no spelling errors in documentation

When the branch is ready to be merged into master:
 - [ ] Update `NEWS.md` with the changes from this pull request under the heading "`# cbioportalR (development version)`". If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [ ] Run `codemetar::write_codemeta()`
- [ ] Run `usethis::use_spell_check()` again
- [ ] Approve Pull Request
- [ ] Merge the PR

### CODE REVIEW

**Comment by hfuchs5** [comment]
I ran checks on this and only the genomic_tbl tests had errors

File: tests/testthat/test-recoding_cna.R [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  expect_no_error(nas_data <- mutate(na_cna, alteration = recode_cna(alteration)))
+  expect_equal(sum(is.na(nas_data$alteration)), 1)
+
+  unk_cna <- gnomeR::cna[1:10, ] %>%
+    rename_columns(.) %>%
+    select(-sample_id) %>% #changing alteration values so drop patient ID
+    mutate(alteration = c(1, "unknown", rep("neutral", nrow(.) - 2)))
+
+  expect_error(mutate(unk_cna, alteration = recode_cna(alteration)))
+
```
Comment (hfuchs5) [inline tests/testthat/test-recoding_cna.R:66]: added an extra test in next commit to test if recode works within sanitize

**Review by hfuchs5 — COMMENTED** [review]
It won't let me approve, but I committed one additional test and I think it looks good!

File: tests/testthat/test-recoding_cna.R (hunk 1 above)
Comment (karissawhiting) [reply tests/testthat/test-recoding_cna.R:66]: great, thank you!

**Pull request merged**